src/stream_management/mod_stream_management.erl

```diff
@@ -148,15 +142,12 @@ session_cleanup(Acc, _LUser, _LServer, _LResource, SID) ->
     mongoose_acc:t().
 do_remove_smid(Acc, HostType, SID) ->
     H = mongoose_acc:get(stream_mgmt, h, undefined, Acc),
-    MaybeSMID = case mnesia:dirty_index_read(sm_session, SID, #sm_session.sid) of
-        [] -> {error, smid_not_found};
-        [#sm_session{smid = SMID}] ->
-            mnesia:dirty_delete(sm_session, SMID),
-            case H of
-                undefined -> ok;
-                _ -> register_stale_smid_h(HostType, SMID, H)
-            end,
-            {ok, SMID}
+    MaybeSMID = unregister_smid(HostType, SID),
+    case MaybeSMID of
```

Comment:
```suggestion
    case unregister_smid(HostType, SID) of
```

Reply:
MaybeSMID is used below the case block though.

Reply:
Ah, didn't notice that xD
Ok so this PR seems perfect for me, can merge 👌🏽